src/components/Login.tsx

```diff
@@ -95,24 +98,23 @@ const Login = ({ isPrimary, loginWalletOptions }: LoginProps): JSX.Element => {
 
   return (
     <div className="Login__block">
-      {!userId ? (
+      {!userId && !registrationPopoverVisible ? (
+        <LoginButton
+          popoverVisible={loginPopoverVisible}
+          setPopoverVisible={setLoginPopoverVisible}
+          loginWalletOptions={loginWalletOptions}
+          loading={loading}
+          loginWithWalletType={login}
+        />
+      ) : (
         <RegistrationModal
-          visible={registrationVisible}
+          visible={registrationPopoverVisible}
+          setRegistrationVisible={setRegistrationPopoverVisible}
           registrations={registrations}
           onIdResolved={setUserID}
-          onCancel={logout}
+          logout={logout}
```

Comment:
now used for cancel button and the logout button, so logout seemed like a better name